src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreArguments.cs

```diff
@@ -0,0 +1,215 @@
+        public static StaticGraphRestoreArguments Read(string[] args)
+        {
+            // Look for the first command-line argument that starts with '@'
+            string argumentFilePath = args.FirstOrDefault(i => i.StartsWith("@", StringComparison.Ordinal))?.Trim('@');
+
+            if (string.IsNullOrWhiteSpace(argumentFilePath))
+            {
+                return null;
+            }
+
+            FileInfo argumentFile = new FileInfo(argumentFilePath);
+
+            if (!argumentFile.Exists)
+            {
+                return null;
+            }
+
+            return Read(argumentFile.FullName);
+        }
+
+        /// <summary>
+        /// Reads arguments from the specified file path.
+        /// </summary>
+        /// <param name="argumentFilePath">The full path to the argument file to read.</param>
+        /// <returns>A <see cref="StaticGraphRestoreArguments" /> object if the specified argument file was valid, otherwise <c>null</c>.</returns>
+        public static StaticGraphRestoreArguments Read(string argumentFilePath)
+        {
+            using FileStream stream = File.OpenRead(argumentFilePath);
+
+            return Read(stream);
+        }
+
+        /// <summary>
+        /// Gets a copy of the global properties.
+        /// </summary>
+        /// <returns>A <see cref="Dictionary{TKey, TValue}" /> containing a copy of the global properties.</returns>
+        public Dictionary<string, string> GetGlobalProperties()
+        {
+            return GlobalProperties.ToDictionary(i => i.Key, i => i.Value, StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Writes the current arguments to the specified file.
+        /// </summary>
+        /// <param name="argumentFilePath">The full path to the arguments file to write. if the file exists, it is overwritten. If the directory does not exist, it will be created.</param>
+        public void Write(string argumentFilePath)
+        {
+            var fileInfo = new FileInfo(argumentFilePath);
+
+            fileInfo.Directory.Create();
+
+            using FileStream stream = File.OpenWrite(argumentFilePath);
+
+            Write(stream);
+        }
+
+        /// <summary>
+        /// Reads arguments from the specified <see cref="Stream" />.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream" /> containing the contents of an argument file.</param>
+        /// <returns>A <see cref="StaticGraphRestoreArguments" /> object if the specified stream contained a valid argument file, otherwise <c>null</c>.</returns>
+        internal static StaticGraphRestoreArguments Read(Stream stream)
+        {
+            using var textReader = new StreamReader(stream);
+            using var reader = new JsonTextReader(textReader);
+
+            while (reader.Read() && reader.TokenType != JsonToken.StartObject)
```

Comment:
Out of curiosity, why not use `JsonSerializer`?

While it is on the hot path, it's only run once per process, so doesn't seem overly performance critical. Less code generally == easier to read/maintain (less risk of bugs, less code changes when new fields added `StaticGraphRestoreArguments`.

Reply:
I decided to use custom logic to maximize speed since this is static graph-based restore and it needs to be as fast as possible.  I felt that serialization was a lot of overhead for this single object which is fairly straightforward.  I felt comfortable having custom logic as long as I had a good unit test and know that only our process reads and writes it so I shouldn't need as much error checking.

Reply:
I decided to try running the benchmark on .NET 6 and with the latest package version:

|                                 Method |                  Job |              Runtime |      InputFile |     Mean |     Error |    StdDev | Ratio | RatioSD |     Gen0 |     Gen1 |  Allocated | Alloc Ratio |
|--------------------------------------- |--------------------- |--------------------- |--------------- |---------:|----------:|----------:|------:|--------:|---------:|---------:|-----------:|------------:|
|      'System.Text.Json JsonSerializer' |             .NET 6.0 |             .NET 6.0 | nuget-org.json | 2.313 ms | 0.0161 ms | 0.0134 ms |  1.00 |    0.00 |  85.9375 |  42.9688 |  536.01 KB |        1.00 |
| 'System.Text.Json with JsonConverters' |             .NET 6.0 |             .NET 6.0 | nuget-org.json | 2.186 ms | 0.0266 ms | 0.0249 ms |  0.94 |    0.01 |  85.9375 |  42.9688 |  528.86 KB |        0.99 |
|       'Newtonsoft.Json JsonSerializer' |             .NET 6.0 |             .NET 6.0 | nuget-org.json | 3.185 ms | 0.0338 ms | 0.0316 ms |  1.38 |    0.02 | 140.6250 |  66.4063 |  800.85 KB |        1.49 |
|  'Newtonsoft.Json with JsonConverters' |             .NET 6.0 |             .NET 6.0 | nuget-org.json | 2.470 ms | 0.0488 ms | 0.0381 ms |  1.07 |    0.02 | 128.9063 |  62.5000 |  801.52 KB |        1.50 |
|                                        |                      |                      |                |          |           |           |       |         |          |          |            |             |
|      'System.Text.Json JsonSerializer' | .NET Framework 4.7.2 | .NET Framework 4.7.2 | nuget-org.json | 4.001 ms | 0.0489 ms | 0.0457 ms |  1.00 |    0.00 |  93.7500 |  46.8750 |  562.52 KB |        1.00 |
| 'System.Text.Json with JsonConverters' | .NET Framework 4.7.2 | .NET Framework 4.7.2 | nuget-org.json | 4.452 ms | 0.0563 ms | 0.0499 ms |  1.11 |    0.01 |  85.9375 |  39.0625 |  551.36 KB |        0.98 |
|       'Newtonsoft.Json JsonSerializer' | .NET Framework 4.7.2 | .NET Framework 4.7.2 | nuget-org.json | 3.673 ms | 0.0492 ms | 0.0436 ms |  0.92 |    0.01 | 152.3438 |  70.3125 |  835.41 KB |        1.49 |
|  'Newtonsoft.Json with JsonConverters' | .NET Framework 4.7.2 | .NET Framework 4.7.2 | nuget-org.json | 2.766 ms | 0.0413 ms | 0.0387 ms |  0.69 |    0.01 | 136.7188 |  66.4063 |  836.25 KB |        1.49 |

So, S.T.J seems to still be slower on netfx than N.J is. Interesting that the custom converts are slower than JsonSerializer's built in converter.

Reply:
I'm a little worried about using System.Text.Json since I'll have to include it in our shipping assemblies for .NET Framework right?

https://apisof.net/catalog/d6e30809-188a-aca1-4cbc-44fd2000ab07

So I'll stick with Newtonsoft.Json for now.